python/tests/plugins/test_gen_plugin.py

```diff
@@ -59,20 +59,23 @@ def test_generate_first_party_plugin(self, is_user_desired_output_dir):
 
     @patch("aac.plugins.gen_plugin.gen_plugin_impl._is_user_desired_output_directory")
     def test_generate_third_party_plugin(self, is_user_desired_output_dir):
+
         with TemporaryDirectory() as temp_directory:
 
-            self.assertEqual(len(os.listdir(temp_directory)), 0)
+            temp_file = temporary_test_file_wo_cm(TEST_PLUGIN_YAML_STRING, dir=temp_directory, suffix=".yaml")
 
-            with temporary_test_file(TEST_PLUGIN_YAML_STRING, dir=temp_directory) as plugin_yaml:
-                is_user_desired_output_dir.return_value = True
-                result = generate_plugin(plugin_yaml.name)
+            is_user_desired_output_dir.return_value = True
+            result = generate_plugin(temp_file.name)
 
-                assert_plugin_success(result)
+            assert_plugin_success(result)
 
-                generated_plugin_files = os.listdir(temp_directory)
-                self.assertEqual(len(generated_plugin_files), 6)
-                self.assertIn(os.path.basename(plugin_yaml.name), generated_plugin_files)
-                self.assertIn(f"{TEST_PLUGIN_NAME}", generated_plugin_files)
+            generated_plugin_files = []
+            for root, dirs, files in os.walk(temp_directory, topdown=False):
+                for file in files:
+                    generated_plugin_files.append(os.path.join(root, file))
```

Comment:
A few things:

1. Comprehensions are a more concise way to build this list.
2. No real need for `topdown=False` here.
3. For variables that aren't used, it's idiomatic to use `_` - so I changed `dirs` to `_`.
4. Looking at the [tests in pipeline run](https://github.com/jondavid-black/AaC/actions/runs/3200458262/jobs/5227412915#step:5:39) I see the following "ignored" exception which seems to indicate that something with these tests are still not working correctly (this also happens locally):
```
Exception ignored in: <function _TemporaryFileCloser.__del__ at 0x7fca0bdf0310>
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/tempfile.py", line 441, in __del__
    self.close()
  File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/tempfile.py", line 437, in close
    unlink(self.name)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp0m5mvk_9/tmp854d6e2p.yaml'
```

```suggestion
            generated_plugin_files = [os.path.join(root, file) for root, _, files in os.walk(temp_directory) for file in files]
```

Reply:
I applied the following above snippet locally and the tests are still failing. 

```......Exception ignored in: <function _TemporaryFileCloser.__del__ at 0x104078f70>
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/tempfile.py", line 445, in __del__
    self.close()
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/tempfile.py", line 441, in close
    unlink(self.name)
FileNotFoundError: [Errno 2] No such file or directory: '/private/var/folders/lr/d4cwjct95f31nlq6vc6pxg7m0000gq/T/tmpkok8m09j/tmp807yvdbf.yaml'
.Exception ignored in: <function _TemporaryFileCloser.__del__ at 0x104078f70>
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/tempfile.py", line 445, in __del__
    self.close()
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/tempfile.py", line 441, in close
    unlink(self.name)
FileNotFoundError: [Errno 2] No such file or directory: '/private/var/folders/lr/d4cwjct95f31nlq6vc6pxg7m0000gq/T/tmp35kim0l0/tmpftn7f26q'
.```

Reply:
To reiterate my comment in stand up, this change was not intended to fix the `FileNotFoundError` but rather it was intended to clean up the creation of the `generated_plugin_files` list.